File: 2021/01.ts
import { readFileSync } from 'fs';

const m = readFileSync('./01input', 'utf-8').split(/\r?\n/).map(Number);
//const m = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263]

var p1 = 0;
for (var i = 0; i < m.length; i++) {
    if (m[i] < m[i+1]) {
        p1++;
    }
}

var p2 = 0;
for (var i = 0; i < m.length-2; i++) {
    var a = m[i] + m[i+1] + m[i+2];
    var b = m[i+1] + m[i+2] + m[i+3];
    if (a < b) {
        p2++;
    }
}

console.log(p1, p2);

File: 2022/01.py
# Using readlines()
file1 = open('2022/01input.txt', 'r', encoding='utf-8')
Lines = file1.readlines()

current_elf = 1
current_calories = 0
calories = {}
for line in Lines:
    line = line.rstrip()
    if line == "":
        #print("next elf")
        calories[current_elf] = current_calories
        current_elf += 1
        current_calories = 0
        continue
    cal = int(line)
    current_calories += cal

calories_values = calories.values()
print("\npart1")
print(max(calories_values))

print("\npart2")
#print(sorted(zip(calories.values(), calories), reverse=True)[:3])
print(sum(sorted(calories_values, reverse=True)[:3]))
